Services/LTI/classes/Screen/LtiViewLayoutProvider.php

```diff
@@ -18,75 +27,116 @@
 class LtiViewLayoutProvider extends AbstractModificationProvider implements ModificationProvider
 {
 
-    /**
-     * @inheritDoc
-     */
-     
     public function isInterestedInContexts() : ContextCollection
     {
         return $this->context_collection->lti();
     }
-    
+
     /**
-     * This is a basic wip implementation for leaving the lti session with an exit button
-     * 
-     * There is an ongoing discussion about a HeaderInfo bar for different ILIAS modes:
-     * 
-     * https://docu.ilias.de/goto_docu_wiki_wpage_5979_1357.html
-     * 
-     * and a PR Feature Discussion:
-     * 
-     * https://github.com/ILIAS-eLearning/ILIAS/pull/2251
-     * 
+     * @inheritDoc
      */
-    public function getMetaBarModification(CalledContexts $screen_context_stack) : ?MetaBarModification
+
+    public function getPageBuilderDecorator(CalledContexts $screen_context_stack) : ?PageBuilderModification
     {
-        $this->dic->logger()->lti()->info("isActive: " . $this->dic["lti"]->isActive());
-        
+        $this->dic->logger()->lti()->info("getPageBuilderDecorator");
         if ($this->dic["lti"]->isActive()) {
             if (isset($_SESSION['lti_launch_css_url']) && $_SESSION['lti_launch_css_url'] != "") {
                 $this->globalScreen()->layout()->meta()->addCss($_SESSION['lti_launch_css_url']);
             }
-            return $this->globalScreen()
-                ->layout()
-                ->factory()
-                ->metabar()
-                ->withModification(function (MetaBar $current) : ?MetaBar {
-                $f = $this->dic->ui()->factory();
-                $close = $f->button()->close();
-                $exit_symbol = $f->symbol()->glyph()->remove();
-                $exit = $f->button()->bulky($exit_symbol,"exit",$this->dic["lti"]->getCmdLink('exit'));
-                $metabar = $f->mainControls()->metaBar()->withAdditionalEntry('exit', $exit);
-                return $metabar;
-            })->withHighPriority();
-        }
-        else {
-            return null;
+            $this->globalScreen()->layout()->meta()->addCss('./Services/LTI/templates/default/lti.css');
```

Comment:
this is not necessarily my business, but why does LTI need its own CSS?

Reply:
I saw this as well, as far as I undwerstood, this seems to be leftover, of some old LTI CSS, right? If not, this seems like an issue.

Reply:
The (small) injected css is just a tiny hack to discard a (null) footer completely. Without an additional display:none the footer is empty but still displayed. In a Kioskmode the footer and btw. other null modifications (Metabar, Mainbar) should hide the components. What do you mean?